test/sqllogictest/transform/reduce_fusion.slt

```diff
@@ -24,7 +24,8 @@ EXPLAIN SELECT DISTINCT r0 FROM (SELECT DISTINCT f1 + 1 as r0, f0 FROM t)
 ----
 %0 =
 | Get materialize.public.t (u1)
-| Distinct group=((#1 + 1))
+| Map (#1 + 1)
+| Distinct group=(#3)
```

Comment:
We can revisit this in the near future, but looks like a case where we want to think through "should MFPs be pushed forward, or absorbed backwards". In this case, pushing into the `Get` makes sense, but only if we had a `Project` there with in. Without the `Project` we would want to absorb it into the `Distinct`. But in this case, I think the absence of the `Project` is the glitch.